.circleci/config.yml

```diff
@@ -2,65 +2,271 @@ version: 2.1
+          name: Restore Pip Cache
+          keys:
+            - pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - pip-packages-v5-{{ .Branch }}-
+            - pip-packages-v5-
       - run:
-          name: lint server (pylint)
+          name: install
           command: |
-            make lint-server
+            make install
+
       - run:
-          name: type check
+          name: Install Chrome
           command: |
-            make typecheck-server
+            curl -L -o google-chrome.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+            sudo dpkg -i google-chrome.deb
+            sudo sed -i 's|HERE/chrome\"|HERE/chrome\" --disable-setuid-sandbox|g' /opt/google/chrome/google-chrome
+            rm google-chrome.deb
       - run:
-          name: check formatting
+          name: create data model
           command: |
-            make format-server
-            git diff-index --quiet HEAD -- || (echo "Found unexpected changes!" && git diff && exit 1)
+            make resetdb
+
       - run:
-          name: server tests
+          name: wdio
           command: |
-            pipenv run pytest -n 2 --cov=.
+            yarn --cwd client test:wdio
+
+  cypress:
+    docker:
+      - image: circleci/python:3.7.3-node-browsers
+        environment:
+          DATABASE_URL: postgresql://root@localhost:5432/arlo-test
+          FLASK_ENV: test
+      - image: circleci/postgres:9.6.3-alpine-ram
+        environment:
+          POSTGRES_USER: root
+          POSTGRES_DB: arlo-test
+    resource_class: xlarge
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          name: Restore Yarn Cache
+          keys:
+            - yarn-packages-v3-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          name: Restore Node Cache
+          keys:
+            - npm-packages-v8-{{ .Branch }}-{{ checksum "package-lock.json" }}
+      - restore_cache:
+          name: Restore Pip Cache
+          keys:
+            - pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - pip-packages-v5-{{ .Branch }}-
+            - pip-packages-v5-
       - run:
-          name: client tests
+          name: install
           command: |
-            make test-client
+            make install
+
       - run:
-          name: wdio end-to-end tests
+          name: Install Chrome
           command: |
-            yarn --cwd client test:wdio
+            curl -L -o google-chrome.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+            sudo dpkg -i google-chrome.deb
+            sudo sed -i 's|HERE/chrome\"|HERE/chrome\" --disable-setuid-sandbox|g' /opt/google/chrome/google-chrome
+            rm google-chrome.deb
```

Comment:
do we need to install chrome separately for the cypress tests (given that the docker image has `browsers` in the name)?